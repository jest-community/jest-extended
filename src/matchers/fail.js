export function fail(_, message) {
  return {
    pass: false,
    message: () => (message ? message : 'fails by .fail() assertion'),
  };
}
