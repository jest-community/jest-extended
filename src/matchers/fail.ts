export function fail(_: unknown, message: string) {
  return {
    pass: false,
    message: () => (message ? message : 'fails by .fail() assertion'),
  };
}
