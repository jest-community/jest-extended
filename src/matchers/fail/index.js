import predicate from './predicate';

const failMessage = message => {
  if (message) return () => message;
  else return () => 'fails by .fail() assertion';
};

export default {
  fail: (expected, message) => ({ pass: predicate(), message: failMessage(message) }),
};
