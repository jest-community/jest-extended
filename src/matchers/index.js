import toBeFalse from './toBeFalse';
import toBeTrue from './toBeTrue';

export default [toBeFalse, toBeTrue].reduce((acc, matcher) => ({ ...acc, ...matcher }), {});
