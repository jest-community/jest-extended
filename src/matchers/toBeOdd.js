export function toBeOdd(received) {
  const { printReceived, matcherHint } = this.utils;

  const passMessage =
    matcherHint('.not.toBeOdd', 'received', '') +
    '\n\n' +
    'Expected value to not be odd received:\n' +
    `  ${printReceived(received)}`;

  const failMessage =
    matcherHint('.toBeOdd', 'received', '') +
    '\n\n' +
    'Expected value to be odd received:\n' +
    `  ${printReceived(received)}`;

  const pass = !isNaN(parseInt(received)) && received % 2 === 1;

  return { pass, message: () => (pass ? passMessage : failMessage) };
}
