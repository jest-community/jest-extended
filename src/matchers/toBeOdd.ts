export function toBeOdd(received) {
  const { printReceived, matcherHint } = this.utils;

  const pass = !isNaN(parseInt(received)) && received % 2 === 1;

  return {
    pass,
    message: () =>
      pass
        ? matcherHint('.not.toBeOdd', 'received', '') +
          '\n\n' +
          'Expected value to not be odd received:\n' +
          `  ${printReceived(received)}`
        : matcherHint('.toBeOdd', 'received', '') +
          '\n\n' +
          'Expected value to be odd received:\n' +
          `  ${printReceived(received)}`,
  };
}
