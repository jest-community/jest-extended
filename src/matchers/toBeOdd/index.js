import { matcherHint, printReceived } from 'jest-matcher-utils';

import predicate from './predicate';

const passMessage = received => () =>
  matcherHint('.not.toBeOdd', 'received', '') +
  '\n\n' +
  'Expected value to not be odd received:\n' +
  `  ${printReceived(received)}`;

const failMessage = received => () =>
  matcherHint('.toBeOdd', 'received', '') +
  '\n\n' +
  'Expected value to be odd received:\n' +
  `  ${printReceived(received)}`;

export default {
  toBeOdd: expected => {
    const pass = predicate(expected);
    if (pass) {
      return { pass: true, message: passMessage(expected) };
    }

    return { pass: false, message: failMessage(expected) };
  }
};
