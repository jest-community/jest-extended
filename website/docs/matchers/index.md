---
sidebar_position: 1
---

# Matchers

- [.pass(message)](/docs/matchers/pass)
- [.fail(message)](/docs/matchers/fail)
- [.toBeEmpty()](/docs/matchers/tobeempty)
- [.toBeOneOf([members])](/docs/matchers/tobeoneof)
- [.toBeNil()](/docs/matchers/tobenil)
- [.toSatisfy(predicate)](/docs/matchers/tosatisfy)

## [Array](/docs/matchers/array)

- [.toBeArray()](/docs/matchers/array/#tobearray)
- [.toBeArrayOfSize()](/docs/matchers/array/#tobearrayofsize)
- [.toIncludeAllMembers([members])](/docs/matchers/array/#toincludeallmembersmembers)
- [.toIncludeAllPartialMembers([members])](/docs/matchers/array/#toincludeallpartialmembersmembers)
- [.toIncludeAnyMembers([members])](/docs/matchers/array/#toincludeanymembersmembers)
- [.toIncludeSameMembers([members])](/docs/matchers/array/#toincludesamemembersmembers)
- [.toPartiallyContain(member)](/docs/matchers/array/#topartiallycontainmember)
- [.toSatisfyAll(predicate)](/docs/matchers/array/#tosatisfyallpredicate)
- [.toSatisfyAny(predicate)](/docs/matchers/array/#tosatisfyanypredicate)
- [.toBeInRange(min, max)](/docs/matchers/array/#tobeinrangemin-max)

## [Boolean](/docs/matchers/boolean)

- [.toBeBoolean()](/docs/matchers/boolean/#tobeboolean)
- [.toBeTrue()](/docs/matchers/boolean/#tobetrue)
- [.toBeFalse()](/docs/matchers/boolean/#tobefalse)

## [Date](/docs/matchers/date)

- [.toBeDate()](/docs/matchers/date/#tobedate)
- [.toBeValidDate()](/docs/matchers/date/#tobevaliddate)
- [.toBeAfter(date)](/docs/matchers/date/#tobeafterdate)
- [.toBeBefore(date)](/docs/matchers/date/#tobebeforedate)
- [.toBeAfterOrEqualTo(date)](/docs/matchers/date/#tobeafterorequaltodate)
- [.toBeBeforeOrEqualTo(date)](/docs/matchers/date/#tobebeforeorequaltodate)
- [.toBeBetween(startDate, endDate)](/docs/matchers/date/#tobebetweenstartdate-enddate)

## [Function](/docs/matchers/function)

- [.toBeFunction()](/docs/matchers/function/#tobefunction)
- [.toThrowWithMessage()](/docs/matchers/function/#tothrowwithmessagetype-message)

## [Mock](/docs/matchers/mock)

- [.toHaveBeenCalledBefore()](/docs/matchers/mock/#tohavebeencalledbefore)
- [.toHaveBeenCalledAfter()](/docs/matchers/mock/#tohavebeencalledafter)
- [.toHaveBeenCalledOnce()](/docs/matchers/mock/#tohavebeencalledonce)
- [.toHaveBeenCalledOnceWith()](/docs/matchers/mock/#tohavebeencalledoncewith)

## [Number](/docs/matchers/number)

- [.toBeNumber()](/docs/matchers/number/#tobenumber)
- [.toBeNaN()](/docs/matchers/number/#tobenan)
- [.toBeFinite()](/docs/matchers/number/#tobefinite)
- [.toBePositive()](/docs/matchers/number/#tobepositive)
- [.toBeNegative()](/docs/matchers/number/#tobenegative)
- [.toBeEven()](/docs/matchers/number/#tobeeven)
- [.toBeOdd()](/docs/matchers/number/#tobeodd)
- [.toBeWithin(start, end)](/docs/matchers/number/#tobewithinstart-end)
- [.toBeInteger()](/docs/matchers/number/#tobeinteger)

## [Object](/docs/matchers/object)

- [.toBeObject()](/docs/matchers/object/#tobeobject)
- [.toBeEmptyObject()](/docs/matchers/object/#tobeemptyobject)
- [.toContainKey(key)](/docs/matchers/object/#tocontainkeykey)
- [.toContainKeys([keys])](/docs/matchers/object/#tocontainkeyskeys)
- [.toContainAllKeys([keys])](/docs/matchers/object/#tocontainallkeyskeys)
- [.toContainAnyKeys([keys])](/docs/matchers/object/#tocontainanykeyskeys)
- [.toContainValue(value)](/docs/matchers/object/#tocontainvaluevalue)
- [.toContainValues([values])](/docs/matchers/object/#tocontainvaluesvalues)
- [.toContainAllValues([values])](/docs/matchers/object/#tocontainallvaluesvalues)
- [.toContainAnyValues([values])](/docs/matchers/object/#tocontainanyvaluesvalues)
- [.toContainEntry([key, value])](/docs/matchers/object/#tocontainentrykey-value)
- [.toContainEntries([[key, value]])](/docs/matchers/object/#tocontainentrieskey-value)
- [.toContainAllEntries([[key, value]])](/docs/matchers/object/#tocontainallentrieskey-value)
- [.toContainAnyEntries([[key, value]])](/docs/matchers/object/#tocontainanyentrieskey-value)
- [.toBeExtensible()](/docs/matchers/object/#tobeextensible)
- [.toBeFrozen()](/docs/matchers/object/#tobefrozen)
- [.toBeSealed()](/docs/matchers/object/#tobesealed)

## [Promise](/docs/matchers/promise)

- [.toResolve()](/docs/matchers/promise/#toresolve)
- [.toReject()](/docs/matchers/promise/#toreject)

## [String](/docs/matchers/string)

- [.toBeString()](/docs/matchers/string/#tobestring)
- [.toBeHexadecimal(string)](/docs/matchers/string/#tobehexadecimal)
- [.toBeDateString(string)](/docs/matchers/string/#tobedatestringstring)
- [.toEqualCaseInsensitive(string)](/docs/matchers/string/#toequalcaseinsensitivestring)
- [.toStartWith(prefix)](/docs/matchers/string/#tostartwithprefix)
- [.toEndWith(suffix)](/docs/matchers/string/#toendwithsuffix)
- [.toInclude(substring)](/docs/matchers/string/#toincludesubstring)
- [.toIncludeRepeated(substring, times)](/docs/matchers/string/#toincluderepeatedsubstring-times)
- [.toIncludeMultiple([substring])](/docs/matchers/string/#toincludemultiplesubstring)
- [.toEqualIgnoringWhitespace(string)](/docs/matchers/string/#toequalignoringwhitespacestring)

## [Symbol](/docs/matchers/symbol)

- [.toBeSymbol()](/docs/matchers/symbol/#tobesymbol)
