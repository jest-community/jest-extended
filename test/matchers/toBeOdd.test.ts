import * as matcher from 'src/matchers/toBeOdd';

expect.extend(matcher);

describe('.toBeOdd', () => {
  test('passes when given an odd number', () => {
    expect(1).toBeOdd();
  });

  test.each([[false], [true], [''], [2], [2n], [{}], [() => {}], [undefined], [null], [NaN], [Symbol('foo')]])(
    'fails when given not given an odd number',
    given => {
      expect(() => expect(given).toBeOdd()).toThrowErrorMatchingSnapshot();
    },
  );
});

describe('.not.toBeOdd', () => {
  test.each([[false], [true], [''], [2], [2n], [[]], [{}], [() => {}], [undefined], [null], [NaN], [Symbol('foo')]])(
    'passes when not given an odd number: %s',
    given => {
      expect(given).not.toBeOdd();
    },
  );

  test('fails when given an odd number', () => {
    expect(() => expect(1).not.toBeOdd()).toThrowErrorMatchingSnapshot();
  });
});
