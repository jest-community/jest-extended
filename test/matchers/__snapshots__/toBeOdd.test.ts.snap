// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeOdd fails when given an odd number 1`] = `
"<dim>expect(</intensity><red>received</color><dim>).not.toBeOdd()</intensity>

Expected value to not be odd received:
  <red>1</color>"
`;

exports[`.toBeOdd fails when given not given an odd number 1`] = `
"<dim>expect(</intensity><red>received</color><dim>).toBeOdd()</intensity>

Expected value to be odd received:
  <red>false</color>"
`;

exports[`.toBeOdd fails when given not given an odd number 2`] = `
"<dim>expect(</intensity><red>received</color><dim>).toBeOdd()</intensity>

Expected value to be odd received:
  <red>true</color>"
`;

exports[`.toBeOdd fails when given not given an odd number 3`] = `
"<dim>expect(</intensity><red>received</color><dim>).toBeOdd()</intensity>

Expected value to be odd received:
  <red>""</color>"
`;

exports[`.toBeOdd fails when given not given an odd number 4`] = `
"<dim>expect(</intensity><red>received</color><dim>).toBeOdd()</intensity>

Expected value to be odd received:
  <red>2</color>"
`;

exports[`.toBeOdd fails when given not given an odd number 5`] = `
"<dim>expect(</intensity><red>received</color><dim>).toBeOdd()</intensity>

Expected value to be odd received:
  <red>{}</color>"
`;

exports[`.toBeOdd fails when given not given an odd number 6`] = `
"<dim>expect(</intensity><red>received</color><dim>).toBeOdd()</intensity>

Expected value to be odd received:
  <red>[Function anonymous]</color>"
`;

exports[`.toBeOdd fails when given not given an odd number 7`] = `
"<dim>expect(</intensity><red>received</color><dim>).toBeOdd()</intensity>

Expected value to be odd received:
  <red>undefined</color>"
`;

exports[`.toBeOdd fails when given not given an odd number 8`] = `
"<dim>expect(</intensity><red>received</color><dim>).toBeOdd()</intensity>

Expected value to be odd received:
  <red>null</color>"
`;

exports[`.toBeOdd fails when given not given an odd number 9`] = `
"<dim>expect(</intensity><red>received</color><dim>).toBeOdd()</intensity>

Expected value to be odd received:
  <red>NaN</color>"
`;
